oak_client/src/attestation.rs

```diff
@@ -91,13 +110,19 @@ impl RemoteAttestationVerifier {
         }
     }
 
-    fn check_tee_report(&self, extension: &str) -> anyhow::Result<()> {
+    /// Check that the [`extension`] contains [`RemoteAttestationVerifier::tee_measurement`] and
+    /// a SHA-256 digest of the [`public_key`].
+    fn check_tee_report(&self, extension: &str, public_key: &[u8]) -> anyhow::Result<()> {
```

Comment:
Is this fn defined on `self` just because it needs access to the expected report value? That does not seem good encapsulation, also since the public key is passed explicitly instead, it is also quite hard to test in isolation. I would expect a top-level method that does the checking perhaps, not tied to a struct impl.